From aee124d143b87107d12e24077bacd3e006ea49b4 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Wed, 12 Sep 2018 18:08:11 +0800
Subject: [PATCH 4625/5242] MLK-19580 net: fec: read MAC address from fuse

commit  09ee88f1eaca98830dc0b1df1c8058a1914fc6c4 from
https://source.codeaurora.org/external/imx/linux-imx.git

uboot fdt_fixup_ethernet() already add the "local-mac-address" property
for each net node when boot from net, the mac address read from fuse in
default. But for non-net boot or net is disabled in uboot, then kernel
also needs to read the MAC address from fuse. The patch add i.MX8MM platform
to read fuse from MAC in kernel in such case.

Reviewed-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/freescale/fec_fixup.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/fec_fixup.c b/drivers/net/ethernet/freescale/fec_fixup.c
index 1781019..d2e2d50 100644
--- a/drivers/net/ethernet/freescale/fec_fixup.c
+++ b/drivers/net/ethernet/freescale/fec_fixup.c
@@ -256,7 +256,8 @@ void fec_enet_get_mac_from_fuse(struct device_node *np, unsigned char *mac)
 	else if (of_machine_is_compatible("fsl,imx8qxp"))
 		imx8qm_get_mac_from_fuse(idx, mac,
 					 &imx8_fuse_mapping[IMX8QXP_FUSE]);
-	else if (of_machine_is_compatible("fsl,imx8mq"))
+	else if (of_machine_is_compatible("fsl,imx8mq") ||
+		 of_machine_is_compatible("fsl,imx8mm"))
 		imx8mq_get_mac_from_fuse(idx, mac);
 }
 
-- 
1.7.9.5

